lib/server/tcp-server.js

```diff
@@ -42,23 +43,24 @@ function Req(msg) {
 function Res(socket, ack) {
   this.ack = ack;
 
-  this.end = function() {
-    socket.write(VT + ack.toString() + FS + CR);
+  this.end = function(overrideAck) {
+    socket.write(VT + (overrideAck || ack).toString() + FS + CR);
   }
 }
 
-TcpServer.prototype.start = function(port, encoding) {
+TcpServer.prototype.start = function(port, encoding, options) {
   var self = this;
+  options = options || {}
   this.server = net.createServer(function(socket) {
     var message = "";
     socket.on('data', function(data) {
       try {
         message += data.toString();
-        if (message.substring(message.length - 2, message.length) == FS + CR) {
+        if (message.substring(message.length - 2, message.length) == FS + CR || options.skipValidation) {
           var hl7 = self.parser.parse(message.substring(1, message.length - 2));
           var ack = self.createAckMessage(hl7);
 
-          var req = new Req(hl7);
+          var req = new Req(hl7, message);
```

Comment:
if you can get by with just this part of the PR (adding the raw underlying message to the req) I will merge that in. 

the skipValidation option seems too non-standard from what I've encountered, so I'd rather no include it.